### PR TITLE
Feat/allow non standard variants of tag parse specific character set while parsing

### PR DESCRIPTION
**Pull request by kanguki**
This PR is a workaround to bypass error ParseSpecificCharacterSet: Unknown character set when we process dicom files having non-standard value for tag SpecificCharacterSet (specifically ISO_2022_IR_6)

### CODE REVIEW

File: parse.go [hunk 1]
```diff
@@ -192,6 +192,9 @@ func (p *Parser) Next() (*Element, error) {
 
 	if elem.Tag == tag.SpecificCharacterSet {
 		encodingNames := MustGetStrings(elem.Value)
+		for i := range encodingNames {
```
Comment (ducquangkstn) [inline parse.go]: The pydicom library is considered a good library, strong community. 
Pls update the code so it looks like the python library.
https://github.com/pydicom/pydicom/blob/d5f9ae9223f4c65a479967e76d6ae5abecefd345/pydicom/charset.py#L721


```python
import pydicom
>>> pydicom.charset.convert_encodings('ISO_2022_IR_6')
/opt/homebrew/lib/python3.11/site-packages/pydicom/charset.py:743: UserWarning: Incorrect value for Specific Character Set 'ISO_2022_IR_6' - assuming 'ISO 2022 IR 6'
  _warn_about_invalid_encoding(encoding, patched)
['iso8859']
```

File: parse.go [hunk 2]
```diff
@@ -222,21 +225,32 @@ type ParseOption func(*parseOptSet)
...
-	allowMismatchPixelDataLength       bool
-	skipPixelData                      bool
-	skipProcessingPixelDataValue       bool
-	allowMissingMetaElementGroupLength bool
+	skipMetadataReadOnNewParserInit                bool
+	allowMismatchPixelDataLength                   bool
+	skipPixelData                                  bool
+	skipProcessingPixelDataValue                   bool
+	allowMissingMetaElementGroupLength             bool
+	tagSpecificCharacterSetToEncodingNameConverter func(string) string
```
Comment (ducquangkstn) [inline parse.go]: To archive maximum flexibility, it makes more sense if the flag output is []*encoding.Decoder

```go
type parseOptSet struct {
   getEncoderFn func([]string) ([]*encoding.Decoder, error)
}

func ParseSpecificCharacterSet(encodingNames []string, fn ...) {
  var decoders []*encoding.Decoder
  var error
  if fn == nil {
     decoders, err = DefaultGetEncoder(encodingNames)
  } else {
    decoders, err = fn(encodingNames)
  }
...
}


```

File: parse.go (hunk 1 above)
Comment (kanguki) [reply parse.go]: added function to make Parser fix common spelling error and use default value in 5b5745ce2222590f196ba3ffc55b7dc845fed76d

File: parse.go (hunk 2 above)
Comment (kanguki) [reply parse.go]: updated at https://github.com/segmed/dicom/pull/2/commits/d90c4f7ebe371515c99a7441551d53f4d3b4d6ec  https://github.com/segmed/dicom/pull/2/commits/60396b4e70cdc3f041b4ef0f0a0592eeda1b3a6e

File: parse.go [hunk 3]
```diff
@@ -206,6 +226,22 @@ func (p *Parser) Next() (*Element, error) {
...
+	defaultSpecificCharacterSet := DefaultEncodingName // corresponds to iso-8859-1 in htmlEncodingNames in pkg/charset
+	fixed := defaultSpecificCharacterSet
+	if match := regexp.MustCompile(`^ISO.IR`).MatchString(specificCharacterSet); match {
+		fixed = "ISO_IR" + specificCharacterSet[6:]
+	} else if match := regexp.MustCompile(`^ISO.2022.IR.`).MatchString(specificCharacterSet); match {
+		fixed = "ISO 2022 IR " + specificCharacterSet[12:]
+	}
+	errLogger := log.New(os.Stderr, "", 0) // avoid altering the output for future use
+	if fixed != specificCharacterSet {
+		errLogger.Printf("Incorrect value for Specific Character Set '%s' - assuming %s", specificCharacterSet, fixed)
```
Comment (ducquangkstn) [inline parse.go]: Pls use `debug.Log` like other places in this code base.

File: parse.go [hunk 4]
```diff
@@ -206,6 +226,22 @@ func (p *Parser) Next() (*Element, error) {
 
 }
 
+// FixCommonSpellingErrorInSpecificCharacterSet references _python_encoding_for_corrected_encoding in https://github.com/pydicom/pydicom/
+func FixCommonSpellingErrorInSpecificCharacterSet(specificCharacterSet string) string {
+	defaultSpecificCharacterSet := DefaultEncodingName // corresponds to iso-8859-1 in htmlEncodingNames in pkg/charset
+	fixed := defaultSpecificCharacterSet
```
Comment (ducquangkstn) [inline parse.go]: I'm confused. Why not just `fixed := DefaultEncodingName`

**Review by ducquangkstn — APPROVED** [review]

File: parse.go (hunk 3 above)
Comment (kanguki) [reply parse.go]: https://github.com/segmed/dicom/pull/2/commits/7579c775bfb505aae6274e34a99da44b3b70963c

File: parse.go (hunk 4 above)
Comment (kanguki) [reply parse.go]: i forgot to change https://github.com/segmed/dicom/pull/2/commits/7579c775bfb505aae6274e34a99da44b3b70963c

**Pull request merged**